src/main/java/org/traccar/protocol/Xexun2ProtocolEncoder.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2022 Stefan Clark (stefan@stefanclark.co.uk)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.protocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.traccar.BaseProtocolEncoder;
+import org.traccar.helper.Checksum;
+import org.traccar.helper.DataConverter;
+import org.traccar.model.Command;
+import org.traccar.Protocol;
+
+public class Xexun2ProtocolEncoder extends BaseProtocolEncoder {
+
+    public Xexun2ProtocolEncoder(Protocol protocol) {
+        super(protocol);
+    }
+
+    public static final int FLAG = 0xfaaf;
+    public static final int MSG_COMMAND = 0x07;
+
+    private static ByteBuf encodeFrame(ByteBuf buf) {
+        int bufLength = buf.readableBytes();
+        if (bufLength < 5) {
+            return null;
+        }
```

Comment:
What is this for?

Reply:
the method would fail if the buffer to encode was < 5 bytes. it should never be the case so could remove these lines.

Reply:
Let's remove if it never happens.